src/main/java/com/exasol/bucketfs/profile/Profile.java

```diff
@@ -9,61 +10,94 @@
  */
 public class Profile {
 
-    public static Profile empty() {
-        return new Profile(null, null, null, null);
+    public static Profile empty(final boolean decodePasswords) {
+        return new Profile(null, null, null, null, null, decodePasswords);
     }
 
     private final String host;
     private final String port;
     private final String bucket;
-    private final String password;
+    private final String readPassword;
+    private final String writePassword;
+    private final boolean decodePasswords;
 
     /**
      * Constructor for productive usage
      *
-     * @param host
-     * @param object
-     * @param bucket
-     * @param password
+     * @param host            host name or IP address of BucketFS service
+     * @param port            port HTTP or HTTPS port the BucketFS service listens on
+     * @param bucket          name of the root bucket
+     * @param readPassword    password for reading, required for private buckets
+     * @param writePassword   password for writing to the bucket
+     * @param decodePasswords {@code true} if BFSC should apply base-64 decoding to passwords from profile or from
+     *                        interactive prompt
      */
-    public Profile(final String host, final String object, final String bucket, final String password) {
+    public Profile(final String host, final String port, final String bucket, final String readPassword,
+            final String writePassword, final boolean decodePasswords) {
         this.host = host;
-        this.port = object;
+        this.port = port;
         this.bucket = bucket;
-        this.password = password;
+        this.readPassword = readPassword;
+        this.writePassword = writePassword;
+        this.decodePasswords = decodePasswords;
     }
 
     /**
-     * @return host
+     * @return host name or IP address of BucketFS service
      */
     public String host() {
         return this.host;
     }
 
     /**
-     * @return port
+     * @return HTTP or HTTPS port the BucketFS service listens on
      */
     public int port() {
         return this.port != null ? Integer.parseInt(this.port) : BucketFsUrl.UNDEFINED_PORT;
     }
 
     /**
-     * @return bucket
+     * @return name of the root bucket
      */
     public String bucket() {
         return this.bucket;
     }
 
     /**
-     * @return password
+     * @return password for reading, required for private buckets
      */
-    public String password() {
-        return this.password;
+    public String readPassword() {
```

Comment:
Method name may be misleading, you could read it as "read the password". Maybe rename to `getReadPassword()` or `getReadingPassword()` or `getPasswordForReading()`. Same for `writePassword()`.

Reply:
OK.